packit/config/package_config.py

```diff
@@ -396,7 +399,14 @@ def get_specfile_path_from_repo(
     :param ref: git ref
     :return: str path of the spec file or None
     """
-    spec_files = project.get_files(ref=ref, filter_regex=r".+\.spec$")
+    try:
```

Comment:
Is this part really required?

At [this line](https://github.com/packit-service/packit-service/pull/589/files#diff-302a5d463449b2a9a7a89fb190e151fdR216) in the packit-service PR, the `spec_file_path` is explicitly set.

But with the changes to `get_package_config_from_repo()` from this pr, which makes `spec_file_path` and optional argument, this could be set in the packit-service PR, so `get_spec_file_path_from_repo()` should never be called.

Am I missing something?

Reply:
Yes, you are right. But I wanted to be explicit and would like to get that warning if someone tries to use it from somewhere else. (The type annotation has `Optional` so it matches the return type.)

Reply:
I can remove that part if you want.

Reply:
The reason I would go for removing it, is that I feel somewhat uncomfortable replacing a `NotImplementedError` with a warning.

The later is much easier to miss, while a `NotImplementedError` is as explicit as it could be: one doesn't even have to read the logs for it too much :)

Reply:
OK, I'll remove it. The main difference is that we fail the whole job in that case, but it can be easily fixed.

Reply:
Fixes.